server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -5597,6 +5616,79 @@ public HypervisorType getHypervisorTypeOfUserVM(long vmId) {
         return userVm.getHypervisorType();
     }
 
+    protected String finalizeUserData(String userData, Long userDataId, VirtualMachineTemplate template) {
+        if (StringUtils.isEmpty(userData) && userDataId == null && (template == null || template.getUserDataId() == null)) {
+            return null;
+        }
+        if (template != null && template.getUserDataId() != null) {
+            switch (template.getUserDataOverridePolicy()) {
+                case denyoverride:
```

Comment:
better to use `UserDataOverridePolicy.denyoverride"

Reply:
This is enum inside an interface, so cannot use that.